tests/load/pipeline/test_snowflake_pipeline.py

```diff
@@ -204,3 +206,73 @@ def test_char_replacement_cs_naming_convention(
     results = rel_.fetchall()
     assert len(results) == 1
     assert "AmlSistUtfoertDato" in rel_.columns_schema
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_sql_configs=True,
+        all_staging_configs=True,
+        with_file_format="parquet",
+        subset=["snowflake"],
+    ),
+    ids=lambda x: x.name,
+)
+def test_snowflake_use_vectorized_scanner(destination_config):
+    """Tests whether the vectorized scanner option is correctly applied when loading Parquet files into Snowflake."""
+
+    def get_copy_into_queries(client: SqlClientBase[Any], table_name: str):
+        """Fetch COPY INTO queries within the last minute containing the table name."""
+        query = """
+        SELECT QUERY_TEXT FROM TABLE(INFORMATION_SCHEMA.QUERY_HISTORY( \
+            END_TIME_RANGE_START => DATEADD(MINUTE, -5, CURRENT_TIMESTAMP), \
+            END_TIME_RANGE_END => CURRENT_TIMESTAMP \
+        )) WHERE QUERY_TEXT LIKE '%COPY INTO%'
+        """
+        return [row[0] for row in client.execute_sql(query) if table_name in row[0]]
+
+    os.environ["DESTINATION__SNOWFLAKE__USE_VECTORIZED_SCANNER"] = "TRUE"
```

Comment:
A general approach: If you are testing a toggle as in this case, I think it makes sense to parametrize this. So you run the same block of code once with the property set to true and once with falls. Of course you will then also have different assertions at the end depending on which version is running, but that way you can ensure that the actual code that runs the pipeline is exactly the same, just once with the flag/toggle switched to on and once to off.